File: Easy/LinkedList/206_ReverseLinkedList.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    struct ListNode* p=NULL;
    struct ListNode* q=NULL;

    while(head!=NULL){
        p= head;
        head=head->next;
        p->next = q;
        q=p;
    }
    return p;
    
}

File: Easy/LinkedList/206_ReverseLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if(not head):
            return head
        p,q = ListNode(), None
        while(head):
            p = head
            head = head.next
            p.next = q
            q=p
        return p